Plugins/Public/base_plugin/CoreModule.cpp

```diff
@@ -575,19 +575,37 @@ bool CoreModule::Timer(uint time)
 float CoreModule::SpaceObjDamaged(uint space_obj, uint attacking_space_obj, float curr_hitpoints, float new_hitpoints)
 {
 	base->SpaceObjDamaged(space_obj, attacking_space_obj, curr_hitpoints, new_hitpoints);
+	
+	if(base->shield_state == PlayerBase::SHIELD_STATE_OFFLINE){
+		// shield offline, return expected damage without modifications
+		return new_hitpoints;
+	}
 
 	if (base->shield_strength_multiplier >= 1.0f || isGlobalBaseInvulnerabilityActive || base->invulnerable == 1)
 	{
 		// base invulnerable, keep current health value
 		return curr_hitpoints;
 	}
-	
-	if(base->shield_state == PlayerBase::SHIELD_STATE_OFFLINE){
-		// shield offline, return expected damage without modifications
-		return new_hitpoints;
+
+	float damageTaken;
+	if (!siegeWeaponryMap.empty())
+	{
+		const auto& siegeDamageIter = siegeWeaponryMap.find(iDmgMunitionID);
+		if (siegeDamageIter == siegeWeaponryMap.end())
+		{
+			//Siege gun(s) defined, but this is not one of them, no damage dealt
+			return curr_hitpoints;
+		}
+		else
+		{
+			//Even with siege gun damage override, it still takes shield strength into the account
+			damageTaken = siegeDamageIter->second * (1 - base->shield_strength_multiplier);
```

Comment:
This one should be moved to a constexpr to denote what it actually means. Also if these are both floats, using an int and not 1.0f is going to raise warnings.

Reply:
base->shield_strength_multiplier is variable, so it can't be condensed into a constant expression. Adjusting the number literals.